packages/cli/src/executions/executions.service.ts

```diff
@@ -225,13 +205,10 @@ export class ExecutionsService {
 		if (!sharedWorkflowIds.length) return false;
 
 		const { id: executionId } = req.params;
-		const execution = await Container.get(ExecutionRepository).findSingleExecution(executionId, {
-			where: {
-				workflowId: In(sharedWorkflowIds),
-			},
-			includeData: true,
-			unflattenData: true,
-		});
+		const execution = (await Container.get(ExecutionRepository).findIfShared(
+			executionId,
+			sharedWorkflowIds,
+		)) as unknown as IExecutionResponse;
```

Comment:
Would be nice if the `findIfShared` returned the correct type for the query. Tho `IExecutionResponse` sounds like something that is an HTTP response, so wouldn't make sense to have that type in the repository. So maybe there is a way to improve the typing here or not.

Reply:
`findIfShared` is a wrapper with hardcoded args for `findSingleExecution` which has multiple overloads, but TS cannot follow that the hardcoded args in the wrapper mean a specific overload, and so TS broadens the output type to all overloads. Ideally I'd love to break up `findSingleExecution` instead to avoid having so many overloads, but that's a bigger refactor.